scripts/tasks/preprocess-api-history.ts

```diff
@@ -0,0 +1,232 @@
+
+const apiHistoryRegex = /<!--[\s\S]*?(```[\s\S]*?```)[\s\S]*?-->/;
+
+interface ChangeSchema {
+  'pr-url': string;
+  'breaking-changes-header'?: string;
+  description?: string;
+}
+
+interface ApiHistory {
+  added?: ChangeSchema[];
+  deprecated?: ChangeSchema[];
+  changes?: ChangeSchema[];
+}
+
+interface LiteralString extends Literal {
+  value: string;
+}
+
+// Copied from here: <https://github.com/electron/website/blob/feat/api-history/scripts/tasks/add-frontmatter.ts#L16-L23>
+const getMarkdownFiles = async (startPath: string) => {
+  const filesPaths = await globby(path.posix.join(startPath, 'api', '/*.md'));
+
+  const files: Map<string, string> = new Map();
+  for (const filePath of filesPaths) {
+    const content = await readFile(filePath, 'utf-8');
+    files.set(filePath, content);
+  }
+
+  return files;
+};
+
+// Copied from here but slightly different: <https://github.com/electron/lint-roller/blob/bac245478ba69017c2a82e8ba1b2884a37647494/bin/lint-markdown-api-history.ts#L42-L67>
+function findPossibleApiHistoryBlocks(tree: Root) {
+  const codeBlocks: Html[] = [];
+
+  visit(
+    tree,
+    (node: Node) =>
+      isHtml(node) &&
+      (node as LiteralString).value.includes('```') &&
+      (node as LiteralString).value.toLowerCase().includes('yaml') &&
+      (node as LiteralString).value.toLowerCase().includes('history'),
+    (node) => {
+      if (!isHtml(node)) return; // Should never happen, but TypeScript needs this check
+      codeBlocks.push(node);
+    }
+  );
+
+  return codeBlocks;
+}
+
+function findValidApiHistoryBlocks(
+  possibleHistoryBlocks: Html[],
+  filePath: string,
+  validateAgainstSchema: ValidateFunction<ApiHistory>
+) {
+  const validHistoryBlocks: Html[] = [];
+
+  // All of the validation logic copied from here: <https://github.com/electron/lint-roller/blob/bac245478ba69017c2a82e8ba1b2884a37647494/bin/lint-markdown-api-history.ts#L117-L176>
+  for (const possibleHistoryBlock of possibleHistoryBlocks) {
+    const {
+      children: [htmlComment],
+    } = fromHtml(possibleHistoryBlock.value);
+
+    if (htmlComment == null) {
+      logger.warn(
+        `Error parsing possible history block (found null/undefined htmlComment) in ${logger.green(
+          filePath
+        )}`
+      );
+      continue;
+    }
+
+    if (htmlComment.type !== 'comment') {
+      logger.warn(
+        `Possible API History block is not in a HTML comment (${logger.green(
+          filePath
+        )})`
+      );
+      continue;
+    }
+
+    const {
+      children: [codeBlock],
+    } = fromMarkdown(htmlComment.value);
+
+    if (codeBlock == null) {
+      logger.warn(
+        `Error parsing possible history block (found null/undefined codeBlock) in ${logger.green(
+          filePath
+        )}`
+      );
+      continue;
+    }
+
+    if (
+      codeBlock.type !== 'code' ||
+      codeBlock.lang?.toLowerCase() !== 'yaml' ||
+      codeBlock.meta?.trim() !== 'history'
```

Comment:
Do need add `toLowerCase`?

Reply:
Whoops, fixed in [`a111aea` (#594)](https://github.com/electron/website/pull/594/commits/a111aeaccb88bcfc0abd44dd923ae8d546292cf7)